Error with GHC 9.0.1

I'm getting this error when trying to build this package with GHC 9.0.1:

```
src/Csound/Dynamic/Types/Exp.hs:202:66: error:
    • No instance for (Data.Functor.Classes.Eq1 RatedExp)
        arising from a use of ‘==’
    • In the second argument of ‘(&&)’, namely
        ‘(ratedExpExp re == EmptyExp)’
      In the expression:
        isNothing (ratedExpDepends re) && (ratedExpExp re == EmptyExp)
      In an equation for ‘isEmptyExp’:
          isEmptyExp e
            = isNothing (ratedExpDepends re) && (ratedExpExp re == EmptyExp)
            where
                re = unFix e
    |
202 | isEmptyExp e = isNothing (ratedExpDepends re) && (ratedExpExp re == EmptyExp)
    |                                                                  ^^
```

See also [this stackoverflow question](https://stackoverflow.com/q/69605645).

--- a/src/Csound/Dynamic/Types/Exp.hs
+++ b/src/Csound/Dynamic/Types/Exp.hs
@@ -387,7 +387,9 @@ instance Hashable a => Hashable (RatedExp a)
 instance Hashable InstrId
 
 deriving instance Generic1 IM.IntMap
+#if !MIN_VERSION_hashable(1,3,4)
 instance Hashable1 IM.IntMap
+#endif
 instance Hashable a => Hashable1 (PreInline a)
 instance Hashable a => Hashable1 (Inline a)
 instance Hashable1 RatedExp

--- a/src/Csound/Dynamic/Types/Exp.hs
+++ b/src/Csound/Dynamic/Types/Exp.hs
@@ -196,10 +196,6 @@ type IsArrInit = Bool
 type ArrSize a = [a]
 type ArrIndex a = [a]
 
-isEmptyExp :: E -> Bool
-isEmptyExp e = isNothing (ratedExpDepends re) && (ratedExpExp re == EmptyExp)
-    where re = unFix e
-
 -- Named variable
 data Var
     = Var
@@ -409,6 +405,10 @@ $(deriveOrd1 ''PreInline)
 $(deriveOrd1 ''Inline)
 $(deriveOrd1 ''MainExp)
 $(deriveOrd1 ''RatedExp)
+
+isEmptyExp :: E -> Bool
+isEmptyExp e = isNothing (ratedExpDepends re) && (ratedExpExp re == EmptyExp)
+    where re = unFix e
 
 --------------------------------------------------------------
 -- comments

